### ISSUE TITLE
goproxytest doesn't support 1.13 cmd/go

### ISSUE THREAD

**Issue by josharian** [rogpeppe/go-internal#84]
Reproduce:

Run the gohack tests using Go 1.13.

Result:

Lots of test failures. The interesting parts look like:

```
            go get rsc.io/sampler@v1.2.1: rsc.io@v1.2.1: reading http://127.0.0.1:60141/mod/rsc.io/@v/v1.2.1.info: 500 Internal Server Error
```

The tests pass with Go 1.12. A cursory glance at the cmd/go source suggests goproxytest changed significantly during Go 1.13, I think for sumdb support in cmd/go.

**Comment by mvdan** [comment]
Interesting... I've been using testscript with goproxytest in a couple of projects, and I've never run into this error with 1.13 or tip. For what it's worth, `go test ./...` on master passes for me with 1.13.1.

**Comment by mvdan** [comment]
gohack master did fail on 1.13, but `go get github.com/rogpeppe/go-internal@latest` fixed it for me by upgrading to v1.4.

**Comment by josharian** [comment]
Indeed! Silly me. Thanks, I'll send a PR to gohack.

(Unfortunately, goproxytest still doesn't support `.../@v/list` urls, but that's a different matter.)
